mypy.ini

```diff
@@ -0,0 +1,553 @@
+[mypy]
+python_version = 3.9
+pretty = true
+exclude = (?x)(
+          docs
+          )
+
+enable_error_code = ignore-without-code
+
+# We can get to passing mypy's strict mode by incrementally changing the
+# following global settings to true. This comes from mypy's documentation.
+
+warn_unused_configs = true
```

Comment:
Right now they are all `true` - you mean you'd change all/some of these to `false` prior to adding this to CI?

Reply:
Good catch!  I was debating back and forth about this.  If we keep all of these true, then we can ensure that new files will be consistent with mypy's strict configuration.  Adding type hint annotations is much easier while writing code, so I'm thinking we should try out strict mode.  If any of the settings cause any headaches, we can set them to be false.

In this case...I need to update the comment!

Reply:
I agree. On the other hand, if you want this in the package I'm in favor of getting it into CI as soon as possible to at least stem the flow of new files without type hints. Maybe we can change some of these to false when adding this to CI to ease into it.